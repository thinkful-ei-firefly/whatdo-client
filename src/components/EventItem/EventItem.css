@import url("https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css");
/* .EventItem__Container {
    width: 95%;
    text-align: center;
    margin-left:auto;
    margin-right:auto;
    align-items: center;
    position: relative;
    min-height: 200px;
} */

.EventItem {
  position: relative;
  display: inline-block;
  padding: 15px 30px;
  /* color: #2196f3; */
  /* text-transform: uppercase; */
  /* letter-spacing: 4px; */
  text-decoration: none;
  /* font-size: 24px; */
  overflow: hidden;
  transition: 0.2s;
  width: 100%;
  border-radius: 7px;
  /* background-color: rgba(255, 255, 255, 0); */
}
.EventItem:hover {
  color: #255784;
  /* background: #2196f3; */
  box-shadow: 0 0 10px #2196f3, 0 0 40px #2196f3;
  transition-delay: 1s;
  border-radius: 7px;
  transform: scale(1.05);
}

.EventItem span {
  position: absolute;
  display: block;
}
.EventItem span:nth-child(1) {
  top: 0;
  left: -100%;
  width: 100%;
  height: 3px;
  background: linear-gradient(90deg, transparent, red);
  box-shadow: 0 0 10px #2196f3;
}
.EventItem:hover span:nth-child(1) {
  left: 100%;
  transition: 1s;
}
.EventItem span:nth-child(3) {
  bottom: 0;
  right: -100%;
  width: 100%;
  height: 3px;
  background: linear-gradient(270deg, transparent, red);
  box-shadow: 0 0 10px #2196f3;
}
.EventItem:hover span:nth-child(3) {
  right: 100%;
  transition: 1s;
  transition-delay: 0.5s;
}
.EventItem span:nth-child(2) {
  top: -100%;
  right: 0;
  width: 3px;
  height: 100%;
  background: linear-gradient(180deg, transparent, red);
  box-shadow: 0 0 10px #2196f3;
}
.EventItem:hover span:nth-child(2) {
  top: 100%;
  transition: 1s;
  transition-delay: 0.25s;
}
.EventItem span:nth-child(4) {
  bottom: -100%;
  left: 0;
  width: 3px;
  height: 100%;
  background: linear-gradient(360deg, transparent, red);
  box-shadow: 0 0 10px #2196f3;
}
.EventItem:hover span:nth-child(4) {
  bottom: 100%;
  transition: 1s;
  transition-delay: 0.75s;
}

.EventItems {
  display: inline;
  /* flex-direction:column ;
      justify-content: center; */
  align-items: center;
  text-align: center;
}

.EventItem__description {
  /* position: relative;
    display: inline-block;
text-decoration: none;
overflow: hidden;
transition: 0.2s;
    width: 100%;
    border-radius: 7px; */
  max-height: 120px;
  display: none;
  overflow: auto;
}
.EventItem__description p {
  /* position: absolute; */
  /* display: block; */
  /* text-align: justify; */
  /* text-align: center; */
  /* z-index: 2; */
}

.EventItem__favorite {
  min-height: 60px;
  margin: 0px;
  padding: 0px;
  font-family: "helvetica Neue";
  position: relative;
}
.button {
  width: 200px;
  height: 50px;
  background-image: radial-gradient(#0048ff, rgb(255, 255, 255));
  position: absolute;
  left: 50%;
  bottom: -160%;
  transform: translate(-50%, -50%);
  border-radius: 3px;
  overflow: hidden;
  cursor: pointer;
  color: white;
  font-size: large;
  font-weight: bold;
}
.button:after,
.button:before {
  content: "Favorite";
  position: absolute;
  left: 50%;
  top: 50%;
  transition: all 0.3s;
  transform: translate(-50%, -50%);
  width: 100%;
}
.button:hover:after {
  top: -50%;
}
.button:before {
  font-family: "Ionicons";
  content: "\f141";
  font-size: 40px;
  top: 190%;
}
.button:hover:before {
  top: 50%;
}

.EventItem:hover .button {
  bottom: -20%;
  transition: 1.5s;
  transition-delay: 1s;
}

.button:hover {
  width: 70px;
  /* transition: .05s; */
  border-radius: 70px;
}

.button:hover {
  /* transition: .05s; */
  /* color:red; */
  /* transition-delay: 1s; */
}

.button:active {
  /* transition: .05s; */
  background: radial-gradient(#ff0000, rgb(255, 255, 255));
  /* transition-delay: 1s; */
}
.button:focus .button:before {
  font-family: "Ionicons";
  content: "\f375";
  font-size: 40px;
  top: 190%;
}

.EventItem:hover .EventItem__description {
  display: block;
}

.EventItem__remove {
  min-height: 60px;
  margin: 0px;
  padding: 0px;
  font-family: "helvetica Neue";
  position: relative;
}
.EventItem__removebtn {
  width: 200px;
  height: 50px;
  background-image: radial-gradient(#ff1900, rgb(255, 255, 255));
  position: absolute;
  left: 50%;
  bottom: -20%;
  transform: translate(-50%, -50%);
  border-radius: 3px;
  overflow: hidden;
  cursor: pointer;
  color: white;
  font-size: large;
  font-weight: bold;
}
.EventItem__removebtn:after,
.EventItem__removebtn:before {
  content: "Remove";
  position: absolute;
  left: 50%;
  top: 50%;
  transition: all 0.3s;
  transform: translate(-50%, -50%);
  width: 100%;
}
.EventItem__removebtn:hover:after {
  top: -50%;
}
.EventItem__removebtn:before {
  font-family: "Ionicons";
  content: "\f252";
  font-size: 40px;
  top: 190%;
}
.EventItem__removebtn:hover:before {
  top: 50%;
}

.EventItem__removebtn:hover {
  width: 70px;
  transition: 0.5s;
  border-radius: 70px;
  /* transition-delay: 1s; */
}

.EventItem:hover .EventItem__removebtn {
  bottom: -20%;
  transition: 1s;
  transition-delay: 1s;
}

.EventItem__removebtn:focus .button:before {
  font-family: "Ionicons";
  content: "\f141";
  font-size: 40px;
  top: 190%;
  background: blue;
}
