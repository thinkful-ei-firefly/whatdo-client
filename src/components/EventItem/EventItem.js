import React from "react";
import TokenService from "../../services/token-service";
import SearchContext from "../../contexts/SearchContext";
import { Link } from "react-router-dom";
import "./EventItem.css";

export default class EventItem extends React.Component {
  static contextType = SearchContext;

  renderFavButton = () => {
    if (!TokenService.hasAuthToken()) {
      //if not logged in, render a "login to save favorites" button that links to the login page
      return (
        <div className="EventItem__goLogin">
          <Link className="button_login" to="/signIn">
            Login to Save Favorites
          </Link>
        </div>
      );
    }
    const inFavorites = this.context.savedEvents.find(
      savedEvent => savedEvent.fetch_id === this.props.fetch_id
    );
    if (inFavorites) {
      //if logged in and favorite already exists in context, render "remove from favorites" button
      return (
        <div className="EventItem__unfavorite">
          <button
            className="EventItem__unfavoritebtn"
            onClick={() => this.context.removeEvent(inFavorites.id)}
          ></button>
        </div>
      );
    }
    //otherwise, render "add to favorites" button
    return (
      <div className="EventItem__favorite">
        <button
          className="button"
          onClick={() => this.context.saveEvent(this.props)}
        ></button>
      </div>
    );
  };

  render() {
    const { name, url, description, venue, address, start_time } = this.props;
    let eventTime = new Date(start_time);
    eventTime = eventTime.toLocaleTimeString();

    return (
      <div className="EventItem__Container">
        <div className="EventItem">
          <span></span>
          <span></span>
          <span></span>
          <span></span>

          <div className="EventItems">
            <h3 className="EventItem__header">{name}</h3>
            {/* <p>
          <img src={imgsrc} alt="thumbnail"/>
        </p> */}
            <h4 className="EventItem__time">
              {venue}, {eventTime}
            </h4>
            <h5 className="EventItem__address">{address}</h5>
            <div className="EventItem__description">
              <p>{description}</p>
            </div>

            <div className="EventItem__learnMore">
              <a href={url} target="_blank" rel="noopener noreferrer">
                Learn More
              </a>
            </div>

            <br></br>
            {this.renderFavButton()}
            {/* <div className='EventItem__favorite'>
  
        <button className='button'  onClick={() => props.saveEvent(props)} ></button>
  
          
        </div> */}
          </div>
        </div>
      </div>
    );
  }
}
//<button onClick={() => props.saveEvent(props)}><i className="fa fa-heart" aria-hidden="true"></i></button>

//      <button className='EventItem__button' onClick={() => props.saveEvent(props)}>Add to favorite<i className="fa fa-heart" aria-hidden="true"></i></button>
