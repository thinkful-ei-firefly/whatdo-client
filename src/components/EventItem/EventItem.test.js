import React from 'react'
import ReactDOM from 'react-dom'
import EventItem from './EventItem'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

describe('<EventItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <EventItem />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <EventItem
            name="Event Name"
            url="https://www.event.com"
            description="An event of events for events by events"
            venue="Event Venue"
            address="9999 Event Avenue, Event Town, Florida 33301"
            start="2019-10-30 17:00:00"
            stop="2019-10-30 23:00:00"
            image={'../../assets/test-images/event-test-image.jpeg'}
          />
        </BrowserRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
