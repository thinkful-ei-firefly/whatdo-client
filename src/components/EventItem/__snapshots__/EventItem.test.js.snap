// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventItem /> renders as expected 1`] = `
<div
  className=".EventItem__Container"
>
  <div
    className="EventItem"
  >
    <span />
    <span />
    <span />
    <span />
    <h3>
      Event Name
    </h3>
    <h4>
      Event Venue
      , 
      5:00:00 PM
    </h4>
    <h5>
      9999 Event Avenue, Event Town, Florida 33301
    </h5>
    An event of events for events by events
     
    <a
      href="https://www.event.com"
    >
      Learn More
    </a>
    <br />
    <button>
      Add to favorites
    </button>
  </div>
</div>
`;
